### PR TITLE
Added phpstan

### LINKED ISSUE #857: Phpstan coverage
Would you accept a PR adding a travis job for phpstan?

... and additional PRs to introduce more specific typehints, e.g. more concrete array return-types etc 

see https://github.com/KnpLabs/php-github-api/blob/94dec1e3753fd9c28aaddba3985db011e445bde0/lib/Github/Api/Issue/Labels.php#L25

This would ease usage of the lib and allow to find errors faster when using it in a upstream project

### PR DESCRIPTION
**Pull request by clxkoders**
closes #857

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -21,6 +21,10 @@ matrix:
     - php: 7.2
       name: Backward compatibillity check
       env: DEPENDENCIES="roave/backward-compatibility-check" TEST_COMMAND="./.github/bc-test.sh"
+    - php: 7.1
+      name: phpstan
+      script:
+        - vendor/bin/phpstan analyse
```
Comment (acrobat) [inline .travis.yml:27]: Add a `--no-progress` here to "clean" the output a bit

File: phpstan-baseline.neon [hunk 2]
```diff
@@ -0,0 +1,3 @@
+parameters:
```
Comment (acrobat) [inline phpstan-baseline.neon:1]: I would remove this file, as we can add the `ignoreErrors` in the main neon file. Also the amount of ignore's should be limited so it should no clutter the phpstan.neon that much

**Comment by acrobat** [comment]
Thank you @clxkoders

File: phpstan-baseline.neon (hunk 2 above)
Comment (staabm) [reply phpstan-baseline.neon]: @acrobat I guess you are not aware of the baseline file mechanics.
The file can automatically be generated by phpstan therefore is a superior approach compared to manually listing ignore rules in the `phpstan.neon.dist`

See https://phpstan.org/blog/phpstans-baseline-feature-lets-you-hold-new-code-to-a-higher-standard

Comment (acrobat) [reply phpstan-baseline.neon]: I did not know this was a feature of phpstan, but would could use it in the future if needed. But for now we can leave this file away as the baseline file is empty for level 0. But thanks for the heads up, something new to learn every day 😄

**Pull request merged**